tinygrad/ops.py

```diff
@@ -152,8 +152,8 @@ def __call__(self, rawbufs:List[RawBuffer], jit=False, force_wait=False) -> Opti
     return et
 
 class Compiled:
-  def __init__(self, buffer: Type[RawBuffer], codegen, runtime, synchronize=lambda: None):
-    self.buffer, self.codegen, self.runtime, self.synchronize = buffer, codegen, runtime, synchronize
+  def __init__(self, buffer: Type[RawBuffer], codegen, runtime, synchronize=lambda: None, device_info = DeviceInfo()):
```

Comment:
why not just use None as the default? should simplify the value extraction in `reduce_op` as you can check if device_info is set or not

Reply:
Good catch, thanks